Terminal.Gui/Core/View.cs

```diff
@@ -596,6 +607,7 @@ public virtual void Remove (View view)
 			if (view == null || subviews == null)
 				return;
 
+			OnRemoving (view);
```

Comment:
You don't use `Removing`. I don't think we should add APIs that there is no clear use case for. We can always add them later once we have a clear use case, but if we get it wrong we've just created bad legacy. 

Side-note: The concern I raised in another comment on `isAdding` illustrates why `Removing` may be more complicated than first imagined.

Reply:
I'm slightly leaning towards keeping Removing() although I cannot think of a single use case. WinForms has this:-


```csharp
// Occurs when a new control is added to the System.Windows.Forms.Control.ControlCollection.
public event ControlEventHandler ControlAdded;

// Occurs when a control is removed from the System.Windows.Forms.Control.ControlCollection.
public event ControlEventHandler ControlRemoved;
```

Reply:
Please remove `Removing` from this PR. I have serious concerns about how it will interact with `IDisposable` and we don't have a clear use case where it's needed. I wish I could explain those concerns in detail, but I'm not confident I understand well enough yet. But I've learned to trust my instincts on things like this. There's no danger in removing the API for now. But there is danger in putting it in and having to change it later. Thanks.

Also, I am arguing for renaming `Adding` to `Added`. I made a comment that argues my point I think pretty clearly. I'll add to that argument that by using `Adding` (current tense`) there's an implication that the operation can be cancelled. However, if we name it `Added` (past tense) it is very clear that the deal is done and there's no going back. 

If you disagree, please explain. If you don't disagree, please rename to `Added`.